### PR TITLE
Fix AwsSecret / Config test suite

### LINKED ISSUE #30: Fix the Secrets Manager test suite

### PR DESCRIPTION
**Pull request by sampierson**
Fixes #30

### CODE REVIEW

File: dcplib/aws_secret.py [hunk 1]
```diff
@@ -17,47 +17,70 @@ class AwsSecret:
...
             self._load()
 
     def _load(self):
         try:
             response = self.secrets_mgr.describe_secret(SecretId=self.name)
             if response:
+                self._debug("AwsSecret.load({}) it exists".format(self.name))
                 self.secret_metadata = response
         except ClientError as e:
+            self._debug("AwsSecret.load({}) ResourceNotFoundException".format(self.name))
```
Comment (kislyuk) [inline dcplib/aws_secret.py]: Well, technically this could be something else... You could just print the type :)

File: dcplib/aws_secret.py [hunk 2]
```diff
@@ -17,47 +17,70 @@ class AwsSecret:
...
 
     def _restore(self):
-        if 'DeletedDate' in self.secret_metadata:
-            self.secrets_mgr.restore_secret(SecretId=self.secret_metadata['ARN'])
-            self._load()
+        self.secrets_mgr.restore_secret(SecretId=self.arn)
+        self._load()
+
+    def _debug(self, message):
+        if self.__class__.debug_logging:
```
Comment (kislyuk) [inline dcplib/aws_secret.py]: Why not omit `.__class__` and just access it directly as `self.debug_logging`?

File: dcplib/aws_secret.py (hunk 1 above)
Comment (sampierson) [reply dcplib/aws_secret.py]: Ah yes. Thanks.

File: dcplib/aws_secret.py (hunk 2 above)
Comment (sampierson) [reply dcplib/aws_secret.py]: 👍

**Pull request merged**